internal/jobservice/repository/sql_job_service.go

```diff
@@ -2,57 +2,55 @@
+	IsJobSetSubscribed(ctx context.Context, queue string, jobSet string) (bool, string, error)
+	UpdateJobServiceDb(ctx context.Context, jobTable *JobStatus) error
+	UpdateJobSetDb(ctx context.Context, queue string, jobSet string, fromMessageId string) error
+	SetSubscriptionError(ctx context.Context, queue string, jobSet string, err string, fromMessageId string) error
+	GetSubscriptionError(ctx context.Context, queue string, jobSet string) (string, error)
+	AddMessageIdAndClearSubscriptionError(ctx context.Context, queue string, jobSet string, messageId string) error
+	UnsubscribeJobSet(ctx context.Context, queue string, jobSet string) (int64, error)
 }
 
 // SQLJobService for persisting to DB.
-type SQLJobService struct {
-	jobServiceConfig *configuration.JobServiceConfiguration
-	db               *sql.DB
-
-	lock sync.RWMutex
-}
-
-func NewSQLJobService(config *configuration.JobServiceConfiguration, db *sql.DB) *SQLJobService {
-	return &SQLJobService{jobServiceConfig: config, db: db}
-}
-
-// Call on a newly created SQLJobService object to setup the DB for use.
-func (s *SQLJobService) Setup() {
-	if s.jobServiceConfig.DatabaseType == "sqlite" {
-		s.useWAL()
-	}
-
-	s.CreateTable()
-}
-
-func (s *SQLJobService) useWAL() {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	_, err := s.db.Exec("PRAGMA journal_mode=WAL")
-	if err != nil {
-		panic(err)
-	}
+type SQLJobService interface {
+	AddMessageIdAndClearSubscriptionError(ctx context.Context, queue string, jobSet string, fromMessageId string) error
+	CheckToUnSubscribe(ctx context.Context, queue string, jobSet string, configTimeWithoutUpdates int64) (bool, error)
+	CleanupJobSetAndJobs(ctx context.Context, queue string, jobSet string) (int64, error)
+	DeleteJobsInJobSet(ctx context.Context, queue string, jobSet string) (int64, error)
+	GetJobStatus(ctx context.Context, jobId string) (*js.JobServiceResponse, error)
+	GetSubscribedJobSets(ctx context.Context) ([]SubscribedTuple, error)
+	GetSubscriptionError(ctx context.Context, queue string, jobSet string) (string, error)
+	HealthCheck(ctx context.Context) (bool, error)
+	IsJobSetSubscribed(ctx context.Context, queue string, jobSet string) (bool, string, error)
+	SetSubscriptionError(ctx context.Context, queue string, jobSet string, connErr string, fromMessageId string) error
+	Setup(ctx context.Context)
+	SubscribeJobSet(ctx context.Context, queue string, jobSet string, fromMessageId string) error
+	UnsubscribeJobSet(ctx context.Context, queue, jobSet string) (int64, error)
+	UpdateJobServiceDb(ctx context.Context, jobTable *JobStatus) error
+	UpdateJobSetDb(ctx context.Context, queue string, jobSet string, fromMessageId string) error
+}
+
+func NewSQLJobService(cfg *configuration.JobServiceConfiguration, log *log.Entry) (SQLJobService, func()) {
+	if cfg.DatabaseType == "postgres" {
+		return NewJSRepoPostgres(cfg, log)
+	} else if cfg.DatabaseType == "sqlite" {
+		return NewJSRepoSQLite(cfg, log)
+	}
+
+	return nil, func() {}
```

Comment:
I thought about that, but adding a third return value - an `error` value - seemed kind of cumbersome.  But I can see that maybe it would avoid confusion if it clearly indicated that a wrong db type was given. I will add that in now.

Reply:
@suprjinx the `NewSQLJobService()` constructor now includes a possible error object, plus the individual Postgres and SQLite implementations are now improved in propagating up errors, not just logging them. Fixed in commit 740a18bf.